### ISSUE TITLE
Redundancy?

### ISSUE THREAD

**Issue by rasbt** [Lightning-AI/litgpt#1408]
I noticed we are using

```
trainable_params = [p for p in model.parameters() if p.requires_grad]
optimizer = AdamW(trainable_params)
```

vs. 

```
optimizer = AdamW(model.parameters())
```

but isn't it the same? I.e., the 2nd would internally filter out all those params with `requires_grad=False`?

Just curious why it was done this 1st way, was there an efficiency advantage?

**Comment by Andrei-Aksionov** [comment]
So, what was the answer?

**Comment by rasbt** [comment]
There shouldn't be a difference :)
